api/api.go

```diff
@@ -122,3 +125,60 @@ func (c *Client) setHeaders() {
 		c.http.SetHeaders(c.config.CustomHTTPHeaders)
 	}
 }
+
+func (c *Client) checkRateLimit(resp *simpleresty.Response) bool {
```

Comment:
According to my Split contacts, the rate limit is, at most, 3 API calls per second. To me, this rate limit seems a bit low so I'm a bit concern a user with lots of resources could have their `terraform plan | apply` take a long time.

Could we add either a max retry count or timeout (configurable) so the user doesn't end up waiting forever? Say default max timeout of 5minutes or 300 seconds?

Reply:
Great, thanks!